gulpfile.js

```diff
@@ -82,14 +82,26 @@ gulp.task('build', ['vendorScripts', 'javascript'], function () {
 // Compiles the user's script files to bundle.js.
 // When including the file in the index.html we need to refer to bundle.js not
 // main.js
-gulp.task('javascript', ['config'], function() {
-  var watcher  = watchify(browserify({
+gulp.task('javascript', function () {
+  // set the correct environment, which controls what happens in config.js
+  if (!process.env.DS_ENV) {
```

Comment:
See the lines below -- in that case, it sets it according to the travis branch / deploy branch stuff.  As you mentioned, we should update it to use PRODUCTION_BRANCH / STAGING_BRANCH, but this will work until we've made that larger change.